src/DataTable/__tests__/DataTable.test.js

```diff
@@ -2794,4 +2794,26 @@ describe('DataTable::direction', () => {
 
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test('should render correctly when sortable column is false', () => {
```

Comment:
I'm not sure this test is correct. We want to setup a case on the table where the snapshot will reflect the state of the table when:

- persistTableHead is true (the headers only show up if this is true)
- progressPending is true
- you cannot sort ie sort is not called:

I would also move these tests to thhe line 492 describe block `describe('when persistTableHead', () => {`

for the test try something like:

```js
  test('should not call onSort if progressPending true', () => {
    const onSortMock = jest.fn();
    const mock = dataMock({ sortable: true });
    const { container } = render(
      <DataTable
        data={mock.data}
        columns={mock.columns}
        onSort={onSortMock}
        persistTableHead
       progressPending
      />,
    );

    fireEvent.click(container.querySelector('div[id="column-some.name"]'));

    expect(onSortMock).not.toBeCalled();
  });
```

Reply:
Actually, with this test a snapshot is not needed

Reply:
Sorry, the test was wrong indeed.

What we want to test is : persistTableHead and no data. We need to be able to click on header column if there is a function bind.
And the sort function must NOT be called, you are right.